derivative.sty

```diff
@@ -251,19 +273,14 @@
 {
 	\keys_define:nn { deriv/dv/#1 }
 	{
-		style-inf    .tl_set:c  = { l__deriv_dv_#1_style_inf_tl },
-		style-var    .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {} },
-		style-var-*  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_star} },
-		style-var-/  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_slash} },
-		style-var-!  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_compact} },
-		style-var-*/  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_star_slash} },
-		style-var-*!  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_star_compact} },
-		style-var-/!  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_slash_compact} },
-		style-var-*/!  .code:n    = { \__deriv_set_style_var:nnn {dv_#1} {##1} {_star_slash_compact} },
```

Comment:
I don't understand why you want to remove the possibility to have distinct styles for `*` variants.
I also don't understand why you want to remove possible styles like `multiple-left`. (But I agree that the styles' names should be improved).

Personally, I think it is better to prepare more possible combinations, even if we do not use them all (so combinations are creating the same results). At worst, users will create new style if they want.